src/thor/isochrone.cc

```diff
@@ -298,6 +312,10 @@ ExpansionRecommendation Isochrone::ShouldExpand(baldr::GraphReader& graphreader,
   // Continue if the time interval has been met. This bus or rail line goes beyond the max
   // but need to consider others so we just continue here. Tells MMExpand function to skip
   // updating or pushing the label back
+
+  // TODO: cost should really also track distance for isodistances to work properly;
+  // right now they'll have an estimated time stop criteria for dijkstra..
+  // that's not the same as shortest
```

Comment:
I _think_ we do need the accumulated distance here from the Cost to say `|| pred.cost().dist > max_distance_`. Apart from a bigger effort regarding changing the Cost definition, this also would mean that both time and distance (if available) would be responsible to stop the dijkstra. If we had another API parameter for contours, e.g. `type` which could specify e.g. `time`, `distance` or `time_distance` (depending on what the user wants the stop criteria to be) we could do smth more flexible.